app/code/Magento/InventoryCatalog/Model/GetLegacyStockItem.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalog\Model;
+
+use Magento\CatalogInventory\Api\Data\StockItemInterface;
+use Magento\CatalogInventory\Api\StockItemCriteriaInterfaceFactory;
+use Magento\CatalogInventory\Model\Stock\Item;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\CatalogInventory\Model\Stock\StockItemRepository;
+
+/**
+ * Service GetLegacyStockItem returns old inventory item data
+ * @package Magento\InventoryCatalog\Model
+ */
+class GetLegacyStockItem
```

Comment:
we no need this service, we need new API for StockItemConfiguration.
Which should be located at InventoryConfigurationApi module

Reply:
Done